src/routes/addresses/gen_multisig_address.rs

```diff
@@ -92,21 +166,110 @@ pub fn validate_address(address: String) -> bool {
         false
     }
 }
+pub async fn service_child_x_pub_key(pool: &PgPool) -> Result<String, sqlx::Error> {
+    let master_keys = get_master_service_keys(&pool).await;
+
+    match master_keys {
+        Ok(master_key) => {
+            let x_pub_key = ExtendedPubKey::from_str(&master_key.master_xpub).unwrap();
+
+            let child_number = x_pub_key.child_number.to_string();
 
-pub async fn service_generated_x_pub_key() -> String {
-    let mnemonic: Mnemonic = keys::generate_mnemonic();
+            let num: u32 = child_number.parse().unwrap();
 
-    let xtended_key: ExtendedKey = keys::generate_extended_key(&mnemonic);
+            let child_pub_key = keys::generate_child_xpub(&x_pub_key, num);
 
-    keys::generate_base58_xpub(xtended_key, Network::Regtest)
+            Ok(child_pub_key.unwrap().to_string())
+        }
+        Err(error) => {
+            println!("{:?}", error);
+            Err(error)
+        }
+    }
+}
+
+pub async fn get_user_derivation_index(
+    user_id: i32,
+    pool: &PgPool,
+) -> Result<DerivationIndex, sqlx::Error> {
+    let derivation_index = sqlx::query_as!(
+        DerivationIndex,
+        r#"
+        select derivation_path from addresses where user_id=$1 ORDER by derivation_path DESC LIMIT 1
+        "#,
+        user_id,
+    )
+    .fetch_one(pool)
+    .await?;
+
+    Ok(derivation_index)
+}
+
+pub async fn get_user_x_pubs(user_email: UserEmail, pool: &PgPool) -> Result<Xpubs, sqlx::Error> {
+    let user_data = sqlx::query_as!(
+        Xpubs,
+        r#"
+            SELECT id, xpub1, xpub2 FROM users WHERE email = ($1)
+            "#,
+        user_email.as_ref(),
+    )
+    .fetch_one(pool)
+    .await?;
+
+    Ok(user_data)
+}
+
+pub async fn get_master_service_keys(pool: &PgPool) -> Result<MasterKeys, sqlx::Error> {
+    let network = option_env!("NETWORK");
+    let keys = sqlx::query_as!(
+        MasterKeys,
+        r#"
+            SELECT master_xpub, master_xpriv FROM service_keys WHERE network=$1 LIMIT 1
+            "#,
+        network,
+    )
+    .fetch_one(pool)
+    .await?;
+
+    Ok(keys)
+}
+
+/// Insert new user to database
```

Comment:
You should update the doc string to be reflective of the function. It is inserting address-related data into the database, and not user data.